### PR TITLE
Filters for booking a ticket

### PR DESCRIPTION
**Pull request by MunteanuM**
intermediary filters for later booking a ticket using htmx and dynamic fields

### CODE REVIEW

**Comment by MunteanuM** [comment]
[[BE] Add the ability to book a ticket](https://trello.com/c/s1IvdvKG/24-be-add-the-ability-to-book-a-ticket)

File: Schedule_app/models.py [hunk 1]
```diff
@@ -6,17 +6,17 @@
...
 class ScheduleMovieCinema(models.Model):
-    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True)
-    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True)
-    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True)
-    hall = models.ForeignKey(CinemaHall, on_delete=models.SET_NULL, null=True)
+    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True, related_query_name='movieback')
+    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True, related_query_name='cityback')
+    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True, related_query_name='cinemaback')
+    hall = models.ForeignKey(CinemaHall, on_delete=models.SET_NULL, null=True, related_query_name='hallback')
     playing = models.DateTimeField(null=True)
```
Comment (maxinasa) [inline Schedule_app/models.py:24]: Normally, a cinema is related to a city and a hall to a cinema. We should not be able to schedule a movie in a hall that is not part of a certain cinema or city, right?

File: cinema/urls.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
     path('users/', include('django.contrib.auth.urls')),
     path('users/', include('app_users.urls')),
     # path('schedule_movie/', include('Schedule_app.urls'))
-    path('book_ticket/', include('Schedule_app.urls'))
+    path('book_ticket/', include('Schedule_app.urls')),
```
Comment (drejleeo) [inline cinema/urls.py:28]: Take a look at this for naming conventions: https://namingconvention.org/python/. In this case the app should be `schedule_app`. I know it's just a detail, but they build up, little by little

File: Schedule_app/views.py [hunk 3]
```diff
@@ -40,22 +40,29 @@ def bookticket(response):
...
-    #         submitted = True
-    # return render(response, 'bookticket.html', {"form": form, 'submitted': submitted, 'logedin': True})
     else:
         return render(response, 'bookticket.html', {'logedin': False})
 
 
 def cinema(response):
     form = TicketBookForm(response.GET)
-    print(form['cinema'])
+    print(response.GET)
```
Comment (drejleeo) [inline Schedule_app/views.py]: It's ok to use `print`s for debugging. But once you finish debugging and prepare your code for a commit, remove any kind of debugging method from the code. Lucky for us this doesn't interrupt the server, but you can never know.

File: Schedule_app/models.py (hunk 1 above)
Comment (drejleeo) [reply Schedule_app/models.py:24]: Make sure to change this also; you don't need so many foreign keys here as some of the are already linked and we can retrieve the city from the cinema and the cinema from the hall as Alex mentioned above

**Review by drejleeo — CHANGES_REQUESTED** [review]

File: cinema/urls.py (hunk 2 above)
Comment (drejleeo) [reply cinema/urls.py:28]: Also, as pretty much all directories are going to host apps it's a bit useless to add the suffix `_app` for the apps.

**Review by drejleeo — CHANGES_REQUESTED** [review]
Add a valid file mask in `.gitignore` for all `.pyc` files to ignore them no matter their location.

File: Schedule_app/views.py [hunk 4]
```diff
@@ -40,22 +44,76 @@ def bookticket(response):
...
+    data = response.POST
+    reservations = []
+    for bookings in data:
+        reservations.append(bookings)
+    reservations.pop(0)
+    reservations.pop(len(reservations) - 1)
+    reservations = SeatModel.objects.filter(name__in=reservations).values_list('id', flat=True)
+    seat_list = ''
+    for seat in reservations:
+        seat_list = seat_list + '{}'.format(seat) + ','
```
Comment (drejleeo) [inline Schedule_app/views.py:112]: Here you can use `','.join(reservations)` instead of that for statement

File: Schedule_app/models.py [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
 
 
 class ScheduleMovieCinema(models.Model):
-    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True)
-    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True)
-    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True)
-    hall = models.ForeignKey(CinemaHall, on_delete=models.SET_NULL, null=True)
+    movie = models.ForeignKey(ScheduledMovies, on_delete=models.SET_NULL, null=True, related_query_name='movieback')
+    city = models.ForeignKey(City, on_delete=models.SET_NULL, null=True, related_query_name='cityback')
+    cinema = models.ForeignKey(Cinema, on_delete=models.SET_NULL, null=True, related_query_name='cinemaback')
```
Comment (drejleeo) [inline Schedule_app/models.py]: I still see no changes here. You still have references for all `city`, `cinema` and `hall`

**Review by drejleeo — CHANGES_REQUESTED** [review]

**Comment by maxinasa** [comment]
Why was this closed like this? It seems that not all the conversations were taken into account. Also, you don't need these changes anymore?

**Pull request closed without merging**